test/autorest/generated/httpinfrastructuregroup/httpredirects.go

```diff
@@ -130,15 +128,23 @@ func (client *httpRedirectsOperations) get300CreateRequest() (*azcore.Request, e
 }
 
 // get300HandleResponse handles the Get300 response.
-func (client *httpRedirectsOperations) get300HandleResponse(resp *azcore.Response) (*HTTPRedirectsGet300Response, error) {
-	if !resp.HasStatusCode(http.StatusOK) {
+func (client *httpRedirectsOperations) get300HandleResponse(resp *azcore.Response) (interface{}, error) {
+	switch resp.StatusCode {
+	case http.StatusOK:
+		result := HTTPRedirectsGet300Response{RawResponse: resp.Response}
```

Comment:
Just for clarity, we want this to be a HTTPRedirectsGet300Response? Or should it be a http.Response?

Reply:
This operation returns a model for HTTP 300, and since 300 aren't automatically redirected we return a model here.

Reply:
I see. OK I was checking because I saw that we're avoiding redirects. But looks good to me if the swagger is returning the model.